sql/src/SqlCsv.sk

```diff
@@ -555,6 +625,8 @@ fun parseTableChange(
   table: SKDB.DirDescr,
   line: String,
 ): ?Array<SKDB.RowValues> {
+  saved = SKStore.newObstack();
```

Comment:
I couldn't convince github to let me suggest changes, so I pushed a commit to your branch with a suggestion to use withRegion here instead of manually saving and restoring the obstack.

Reply:
Oh, could probably use that in the other places too. I might have misunderstood: I thought it only copied out the context.

Reply:
It does something quite similar to what you had, but also protects against exceptions, etc.:
```
fun withRegion<T>(
  context: mutable Context,
  f: (Obstack, mutable Context) ~> T,
): T {
  optSaved: ?Obstack = None();
  try {
    !optSaved = Some(newObstack());
    saved = optSaved.fromSome();
    fresult = f(saved, context);
    copied = destroyObstackWithValue(saved, List[(context.clone(), fresult)]);
    (icontext, copiedResult) = copied.getHead();
    context.replaceFromSaved(icontext);
    copiedResult
  } catch {
  | exn ->
    optSaved match {
    | Some(saved) ->
      cexn = destroyObstackWithValue(saved, List[exn]).getHead();
      throw cexn
    | _ -> throw exn
    }
  }
}
```

Reply:
`withRegion` does something different in the `destroyObstackWithValue` call, passing in the context and then later calling `replaceFromSaved`. I haven't fully understood the consequences of that difference yet.